libs/payments/paypal/src/lib/paypal.manager.ts

```diff
@@ -43,4 +50,58 @@ export class PayPalManager {
     const response = await this.client.setExpressCheckout({ currencyCode });
     return response.TOKEN;
   }
+
+  /**
+   * Process an invoice when amount is greater than minimum amount
+   *
+   * @param customer
+   * @param invoice
+   * @param batchProcessing
+   * @param ipaddress
+   */
+  async processNonZeroInvoice(
+    customer: Stripe.Customer,
+    invoice: Stripe.Invoice,
+    batchProcessing?: boolean,
+    ipaddress?: string
+  ) {
+    // TODO in M3b: Implement legacy processInvoice as processNonZeroInvoice here
+    // TODO: Add spec
+    console.log(customer, invoice, batchProcessing, ipaddress);
+  }
+
+  /**
+   * Finalize and process a draft invoice that has no amounted owed.
+   *
+   * @param invoice
+   */
+  async processZeroInvoice(invoiceId: string) {
+    // It appears for subscriptions that do not require payment, the invoice
+    // transitions to paid automatially.
+    // https://stripe.com/docs/billing/invoices/subscription#sub-invoice-lifecycle
+    return this.stripeManager.finalizeInvoiceWithoutAutoAdvance(invoiceId);
+  }
+
+  /**
+   * Process an invoice
+   * If amount is less than minimum amount, call processZeroInvoice
+   * If amount is greater than minimum amount, call processNonZeroInvoice (legacy PaypalHelper processInvoice)
+   *
+   * @param invoice
+   */
+  async processInvoice(invoice: Stripe.Invoice) {
+    const amountInCents = invoice.amount_due;
+
+    if (amountInCents < this.stripeManager.getMinimumAmount(invoice.currency)) {
+      return await this.processZeroInvoice(invoice.id);
+    } else {
+      const customer = invoice.customer as Stripe.Customer;
```

Comment:
Can we assume that invoice.customer will be expanded?

From what I recall, and the decision in the 0040-stripe-typings ADR, we won't use expanded properties at all. So if the customer is required, it'd have to be fetched from Stripe.